src/scheduler/pbs_sched.c

```diff
@@ -786,11 +780,18 @@ read_sched_cmd(int sock)
 		 * since we are not yet in middle of schedule cycle
 		 */
 		cmd_prio = new_sched_cmd();
-		if (cmd_prio == NULL)
-			return 1; /* return 1 as we have enqueued first command */
-
-		get_sched_cmd_noblk(sock, cmd_prio);
-		free_sched_cmd(cmd_prio);
+		if (cmd_prio != NULL) {
+			int rc_prio = get_sched_cmd_noblk(sock, cmd_prio);
+			free_sched_cmd(cmd_prio);
+			if (rc_prio == -2) {
+				free_sched_cmd(cmd);
+				return 0;
```

Comment:
This means the secondary connection has dropped.  It doesn't necessarily mean the primary has.  Could we gt into a problem where we only have one connection?

Reply:
No, if any of connection is dropped, both (server and sched) will close other open connection, and sched will reopen both connection fresh.